chapters/classes.tex

```diff
@@ -409,14 +409,18 @@ \subsection{Conditional Component Declaration}\label{conditional-component-decla
 \cref{interface-compatibility-or-subtyping}).
 
 If the \lstinline!Boolean! expression is false the component (including its modifier) is removed from the flattened DAE, and connections to/from the component are removed.
-A component declared with a condition-attribute can only be modified and/or used in connections.
+Such a component can only be modified, used in connections, and/or used in a modifier of another conditional component with a false condition.
+
+There are no restrictions on the component if the \lstinline!Boolean! expression is true.
 
```

Comment:
Is it valid to reference such components in equations or algorithms if the condition is tested to be true beforehand? E.g.
```
...
equation
  if  level==1 then
    component1.u =0;
...
```

Reply:
If yes, it would be helpful, to have an operator to check whether the condition on a component is true or not. This would ease the use of libraries having this feature and where the user does not know (because of encryption) or is not able to build up the same condition (protected elements used), e.g.
```
equation
...
    if isDeclared(component1) then
        component1.u=0;
...
```

Reply:
> Is it valid to reference such components in equations or algorithms if the condition is tested to be true beforehand? E.g.
> 
> ```
> ...
> equation
>   if  level==1 then
>     component1.u =0;
> ...
> ```
Since there are no restrictions on using it when enabled it would be legal when level==1, but illegal otherwise - which clearly is not a good model.

I agree that having some way of checking if the component is declared or otherwise forcing evaluation of if-conditions would ease the use of this, but I think we can discuss this after this PR - as the current situation is too problematic.

Reply:
I would write 
`It is an error if conditionally declared components whose declaration condition is evaluated to false during compilation could potentially be used in the simulation model.`

Reply:
> I would remove it and write instead `It is an error if conditionally declared components whose declaration condition is evaluated to false during compilation could potentially be used in the simulation model.`

To me that misses the point that connect-statements involving them are legal (the connection is removed if the condition is false); as it is hard to claim that connect doesn't use them; and similarly for a modifier of them.

Thus I still think the current text is good enough as a first step correcting the currently most problematic cases, and we can later refine it.

Reply:
Regarding:

> Isn't it true, that parameters referenced in conditions of conditional declared components have to be evaluated anyway? So, if in other condition expressions the same parameters occur, the conditions can be evaluated, and the invalid equations in the body are removed.

I would still wait with this for some reasons:
- Previously we haven't required that if-bodies with a condition that can be evaluated to false must be removed, only that isn't run. I'm not sure how that will impact models.
- There are some issues with the order of evaluation
- A different possibility that exist in some languages (e.g., recent C++) is to have a separate if-variant for parts that must be compile-time evaluated